### ISSUE TITLE
Investigate latency spike in 100 node scale test

### ISSUE THREAD

**Issue by bprashanth** [kubernetes/kubernetes#19773]
Looks like we're taking a really long time to store certain objects, sometimes:
```
I0117 00:22:07.677482       5 trace.go:57] Trace "Create /api/v1/namespaces/e2e-tests-density-thzqo/events" (started 2016-01-17 00:22:01.170871168 +0000 UTC):
[12.419µs] [12.419µs] About to convert to expected version
[283.979µs] [271.56µs] Conversion done
[298.602µs] [14.623µs] About to store object in database
[6.506386248s] [6.506087646s] Object stored in database
[6.50639953s] [13.282µs] Self-link added
[6.506526716s] [127.186µs] END
```

And from the test:
```
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/density.go:130
Jan 16 16:25:00.240: INFO: WARNING Top latency metric: {Resource:pods Verb:POST Latency:{Perc50:112.364ms Perc90:1.201764s Perc99:6.469796s}}
Jan 16 16:25:00.240: INFO: WARNING Top latency metric: {Resource:pods Verb:LIST Latency:{Perc50:198.027ms Perc90:950.88ms Perc99:5.522322s}}
Jan 16 16:25:00.240: INFO: Top latency metric: {Resource:nodes Verb:PUT Latency:{Perc50:35.264ms Perc90:113.387ms Perc99:624.519ms}}
Jan 16 16:25:00.240: INFO: Top latency metric: {Resource:pods Verb:PUT Latency:{Perc50:33.038ms Perc90:90.131ms Perc99:590.699ms}}
Jan 16 16:25:00.240: INFO: Top latency metric: {Resource:secrets Verb:GET Latency:{Perc50:11.95ms Perc90:57.214ms Perc99:541.275ms}}
Jan 16 16:25:00.240: INFO: API calls latencies: {
```
https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/logs/kubernetes-e2e-gce-scalability/4046/artifacts/

We already increased threshold to 1s so this is either a gce latency problem or a legitimate regression.  @kubernetes/goog-control-plane

**Comment by timothysc** [comment]
@bprashanth please /cc @kubernetes/sig-scalability on these issues.

**Comment by hongchaodeng** [comment]
We have observed similar spikes caused by GC before. API server has been using unreasonable amount of memory -- double the size of etcd.

> https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/logs/kubernetes-e2e-gce-scalability/4046/artifacts/

We can't access the link. @bprashanth @gmarek Can you share more information such as what to run to reproduce, logs, etc.

We can help look into it. If it's related to GC spike, the right thing to do is decrease memory footprint.

/cc @xiang90

**Comment by timothysc** [comment]
> We have observed similar spikes caused by GC before.

Yup, this is part of the reason why we were pushing to get golang 1.5 in for 1.2 

> API server has been using unreasonable amount of memory -- double the size of etcd.

Do we have a profile to determine what you mean by "unreasonable"

**Comment by hongchaodeng** [comment]
> Do we have a profile to determine what you mean by "unreasonable"

In #15064, there is discussion on memory footprint. I'm planning to have more detailed profile by this quarter.

The thought on top of my head is that api server just keeps a cache of etcd, but it uses twice memory than etcd. This sounds unreasonable. If anyone has thoughts, let me know.

**Comment by gmarek** [comment]
I was caused by the update in go version to 1.4.3 fixed by #19847
